diff --git a/felix.pages.inc b/felix.pages.inc
index 2369920..69eff1a 100644
--- a/felix.pages.inc
+++ b/felix.pages.inc
@@ -243,7 +243,7 @@ function felix_add_node_form($form, &$form_state, $region, $nodetype, $path, $vi
     '#required' => TRUE,
   );
 
-  if (module_exists('linkit')) {
+  if (module_exists('linkit')) { /*
     // Check which version of linkit is currently installed.
     $added = FALSE;
     foreach (array('bac', 'linkit_bac') as $library) {
@@ -261,7 +261,7 @@ function felix_add_node_form($form, &$form_state, $region, $nodetype, $path, $vi
       $form['node']['#prefix'] = '<div id="felix-search-node">';
       $form['node']['#suffix'] = '</div>';
     }
-  }
+  */ }
 
   $form['submit'] = array(
     '#type' => 'submit',
